kedro/extras/datasets/matplotlib/matplotlib_writer.py

```diff
@@ -144,9 +152,9 @@ def _save(
         if (
             isinstance(data, (list, dict))
             and self._overwrite
-            and self._fs.exists(self._filepath)
+            and self._fs.exists(save_path)
```

Comment:
```suggestion
            and self._fs.exists(get_filepath_str(save_path, self._protocol))
```
OR
```suggestion
            and self._exists()
```
since we now know that if `overwrite` is enabled, versioning can't be and hence `self._filepath == self._get_save_path() == self._get_load_path()`.